### PR TITLE
close #68 finish bi request specs.

### LINKED ISSUE #68: Finish Business Intelligence controller tests.

### PR DESCRIPTION
**Pull request by pwentz**

### CODE REVIEW

File: spec/support/customers_sample_data_helper.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    merchant_1, merchant_2 = create_list(:merchant, 2)
+    3.times do |i|
+      if i < 2
+        invoice =
+          create(:invoice, customer: customer, merchant: merchant_1)
+        create(:transaction, invoice: invoice)
+      else
+        invoice =
+          create(:invoice, customer: customer, merchant: merchant_2)
+        create(:transaction, invoice: invoice)
```
[the remark this answers is not in the document]
Comment (pwentz) [reply spec/support/customers_sample_data_helper.rb:13]: ignoring.

File: spec/support/customers_sample_data_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module CustomersSampleDataHelper
+  def favorite_merchant_sample_data
+    customer = create(:customer)
+    merchant_1, merchant_2 = create_list(:merchant, 2)
+    3.times do |i|
+      if i < 2
+        invoice =
+          create(:invoice, customer: customer, merchant: merchant_1)
+        create(:transaction, invoice: invoice)
```
[the remark this answers is not in the document]
Comment (pwentz) [reply spec/support/customers_sample_data_helper.rb:9]: ignoring.

[the remark this answers is not in the document]
Comment (ryanflach) [reply spec/support/customers_sample_data_helper.rb:9]: This was previously in my file - I had this behavior there intentionally.

File: spec/models/customer_spec.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 require 'rails_helper'
-include SampleDataHelper
+include CustomersSampleDataHelper
```
Comment (ryanflach) [inline spec/models/customer_spec.rb:2]: Really like the refactoring into more specific modules - much more readable.

File: spec/models/merchant_spec.rb [hunk 4]
```diff
@@ -9,15 +10,20 @@
...
 
   it 'returns the top ranked x merchants by total items sold' do
-    merchants = top_merchants
+    merchants = top_three_merchants_by_items_sold
+    excluded_merchant = merchants.last
 
     expect(
       Merchant.most_items_sold(2)
-    ).to eq([merchants.first, merchants.last])
+    ).to eq(merchants[0..1])
```
Comment (ryanflach) [inline spec/models/merchant_spec.rb:18]: Cool use of chaining the `.to eq` here - didn't know you could do that.

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -1,13 +1,5 @@
...
-SimpleCov.start do
-  add_filter '/spec/'
-  add_filter '/config/'
-
-  add_group "Controllers", "app/controllers"
-  add_group "Models", "app/models"
-  add_group "Helpers", "app/helpers"
-  add_group "Services", "app/services"
-end
+SimpleCov.start "rails"
```
Comment (ryanflach) [inline spec/spec_helper.rb:2]: Good change - I think this will provide more robust coverage.

**Comment by ryanflach** [comment]
@pwentz Great work on this - merging.

**Pull request merged**